workflow/rules/utils.smk

```diff
@@ -33,4 +33,29 @@ rule tabix_known_variants:
         get_tabix_params
     cache: True
     wrapper:
-        "0.45.1/bio/tabix"
\ No newline at end of file
+        "0.45.1/bio/tabix"
+
+rule bam_regions:
+    input:
+        ref="resources/genome.fasta",
+        bam="results/recal/{sample}.sorted.bam"
+    output:
+        temp("results/regions/{sample}.bed")
+    log:
+        "logs/bam-regions/{sample}.log"
+    conda:
+        "../envs/bedregions.yaml"
+    shell:
+        "samtools view -b {input.bam} | genomeCoverageBed -ibam stdin -g {input.ref} > {output} 2> {log}"
+
+rule merge_regions:
+    input:
+        get_group_beds
+    output:
+        "results/regions/{group}.bed"
+    log:
+        "logs/merge-regions/{group}.log"
+    conda:
+        "../envs/bedtools.yaml"
+    shell:
+        "cat {input} | sort -k1,1 -k2,2n | bedtools merge -i stdin > {output} 2> {log} "
```

Comment:
I think the files can be given as direct arguments to sort, no cat needed. Do you have a source explaining why sort is needed here? Can't bedtools do this on its own?

Reply:
Actually the bedtools documentation tells that data needs to be presorted.
The the "Note" at the[documentation](https://bedtools.readthedocs.io/en/latest/content/tools/merge.html)